### PR TITLE
firebreak: use new signature validation support classes

### PR DESCRIPTION
**Pull request by chrisholmes**
Also, includes use of the MetadataResolveBundle for contructing the
Verify and MSA `MetadataResolver` instances (and trust engine and
credential resolver).

Deleted `DropwizardMetadataResolverFactory` and `MetadataPublicKeyExtractor`

### CODE REVIEW

**Comment by IreneLau-GDS** [comment]
Looks okay to me. Just have a couple of comments from my super OCD...

File: build.gradle [hunk 1]
```diff
@@ -14,7 +14,7 @@ project.ext {
     version_number = '1.0.0'
     openSamlVersion = '3.3.0'
     verifyCommonUtils = '2.0.0-326'
-    samlLibsVersion = "$openSamlVersion-104"
+    samlLibVersion = "$openSamlVersion-107"
```
Comment (IreneLau-GDS) [inline build.gradle]: Open SAML libs have moved on now... 126

Comment (chrisholmes) [reply build.gradle]: maybe, but that's out of scope for this PR

Comment (chrisholmes) [reply build.gradle]: we can raise another PR about that later

File: src/main/java/uk/gov/ida/verifyserviceprovider/factories/VerifyServiceProviderFactory.java [hunk 2]
```diff
@@ -27,26 +28,25 @@
...
-    private final DropwizardMetadataResolverFactory metadataResolverFactory = new DropwizardMetadataResolverFactory();
-    private final Environment environment;
     private final VerifyServiceProviderConfiguration configuration;
     private final ResponseFactory responseFactory;
 
-    private volatile MetadataResolver hubMetadataResolver;
-    private volatile MetadataResolver msaMetadataResolver;
     private final DateTimeComparator dateTimeComparator;
     private final EntityIdService entityIdService;
+    private MetadataResolverBundle verifyMetadataBundler;
```
Comment (IreneLau-GDS) [inline src/main/java/uk/gov/ida/verifyserviceprovider/factories/VerifyServiceProviderFactory.java]: These should be final

Comment (chrisholmes) [reply src/main/java/uk/gov/ida/verifyserviceprovider/factories/VerifyServiceProviderFactory.java]: yes!

Comment (IreneLau-GDS) [reply src/main/java/uk/gov/ida/verifyserviceprovider/factories/VerifyServiceProviderFactory.java]: So are you going to make this final? 😉

File: src/main/java/uk/gov/ida/verifyserviceprovider/factories/EncrypterFactory.java [hunk 3]
```diff
@@ -1,19 +1,21 @@
...
 import org.opensaml.saml.saml2.encryption.Encrypter;
-import org.opensaml.security.credential.BasicCredential;
-import uk.gov.ida.verifyserviceprovider.metadata.MetadataPublicKeyExtractor;
+import org.opensaml.security.credential.Credential;
+import uk.gov.ida.saml.security.EncryptionCredentialResolver;
 
 public class EncrypterFactory extends uk.gov.ida.saml.security.EncrypterFactory {
 
-    private final MetadataPublicKeyExtractor metadataPublicKeyExtractor;
+    private EncryptionCredentialResolver encryptionCredentialResolver;
```
Comment (IreneLau-GDS) [inline src/main/java/uk/gov/ida/verifyserviceprovider/factories/EncrypterFactory.java]: Oh these can be final too.

**Pull request merged**